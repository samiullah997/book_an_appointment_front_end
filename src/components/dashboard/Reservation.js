import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';

import { addReservation } from '../../redux/reducer/reservationReducer';

const Reservation = () => {
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const bikeData = useSelector((state) => state.bikeReducer);
  const { bike } = bikeData;
  const reservationData = useSelector((state) => state.reservationReducer);
  const { reservations } = reservationData;
  const userData = localStorage.getItem('bookBikeUser');
  const user = JSON.parse(userData);
  if (!user) {
    navigate('/');
  }
  const submitData = (e) => {
    e.preventDefault();
    const date = document.getElementById('date').value;
    const data = {
      user_id: user.userId,
      bike_id: bike.id,
      date,
      reserve: true,
    };
    dispatch(addReservation(data));
    document.getElementById('date').value = '';
    navigate('/user/dashboard');
  };

  const checkReservation = (id) => {
    const check = reservations.find((reservation) => reservation.bike_id === id);
    if (check) {
      return true;
    }
    return false;
  };

  return (
    <div className="w-full h-full bg-image border border-black">
      <form className="w-full h-full" onSubmit={submitData}>
        <div className="flex flex-col w-full h-full space-x-3 px-3 py-3 bg-green-700 bg-opacity-70 justify-center items-center">
          <img src={bike.bike_image} alt="bike" className="w-3/4 sm:w-1/4  mt-4" />
          <h1 className="font-bold text-2xl py-5 text-white">{bike.name}</h1>
          <hr />
          <div className="flex flex-col w-full h-full justify-center items-center">
            <input type="date" required name="date" id="date" className="bg-green-500 mt-4 w-3/4 sm:w-1/4 hover:bg-green-700 text-white font-bold py-2 px-4 rounded-full" />
            {checkReservation(bike.id) ? (<img src="https://media.istockphoto.com/id/828928236/vector/red-rubber-stamp-icon-on-transparent-background.jpg?b=1&s=612x612&w=0&k=20&c=gP3E8Zi5YAaKcqdH7ZOsrDdNzHURvQ2gJNdZORjH9PA=" className="h-10 w-30" alt="reserved" />) : <button type="submit" className="bg-green-500 mt-4 hover:bg-green-700 text-white font-bold py-2 px-4 rounded-full">Reserve</button>}

          </div>
        </div>
      </form>
    </div>
  );
};
export default Reservation;
