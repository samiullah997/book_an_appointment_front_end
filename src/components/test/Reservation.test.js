import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../redux/Store';
import Reservation from '../dashboard/Reservation';

describe('Reservations', () => {
  it('renders correctly', () => {
    window.localStorage.setItem('bookDoctorUser', 1);
    const tree = render(<Provider store={store}><Router><Reservation /></Router></Provider>);
    expect(tree).toMatchSnapshot();
  });
});
