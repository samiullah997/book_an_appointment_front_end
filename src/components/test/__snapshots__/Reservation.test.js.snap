// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reservations renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full h-full bg-image border border-black"
      >
        <form
          class="w-full h-full"
        >
          <div
            class="flex flex-col w-full h-full space-x-3 px-3 py-3 bg-green-700 bg-opacity-70 justify-center items-center"
          >
            <img
              alt="bike"
              class="w-3/4 sm:w-1/4  mt-4"
            />
            <h1
              class="font-bold text-2xl py-5 text-white"
            />
            <hr />
            <div
              class="flex flex-col w-full h-full justify-center items-center"
            >
              <input
                class="bg-green-500 mt-4 w-3/4 sm:w-1/4 hover:bg-green-700 text-white font-bold py-2 px-4 rounded-full"
                id="date"
                name="date"
                required=""
                type="date"
              />
              <button
                class="bg-green-500 mt-4 hover:bg-green-700 text-white font-bold py-2 px-4 rounded-full"
                type="submit"
              >
                Reserve
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full h-full bg-image border border-black"
    >
      <form
        class="w-full h-full"
      >
        <div
          class="flex flex-col w-full h-full space-x-3 px-3 py-3 bg-green-700 bg-opacity-70 justify-center items-center"
        >
          <img
            alt="bike"
            class="w-3/4 sm:w-1/4  mt-4"
          />
          <h1
            class="font-bold text-2xl py-5 text-white"
          />
          <hr />
          <div
            class="flex flex-col w-full h-full justify-center items-center"
          >
            <input
              class="bg-green-500 mt-4 w-3/4 sm:w-1/4 hover:bg-green-700 text-white font-bold py-2 px-4 rounded-full"
              id="date"
              name="date"
              required=""
              type="date"
            />
            <button
              class="bg-green-500 mt-4 hover:bg-green-700 text-white font-bold py-2 px-4 rounded-full"
              type="submit"
            >
              Reserve
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
